// promises-utils.wait/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.wait = exports.UNITS = void 0;
/**
 * Instead of passing a String literal to the timeUnit of the wait function options you can use this constant
 */
exports.UNITS = {
    MS: 'ms',
    S: 's',
    M: 'm',
    H: 'h',
    D: 'd',
};
/**
 * Accepts a given time from the point of being called and resolves after that time
 * @param {number} time The amount of time to wait before finishing
 * @param {WaitOptions} waitOptions
 * @returns A Promise that resolves after the given time
 */
function wait(time, _a) {
    var _b = _a === void 0 ? {} : _a, _c = _b.timeUnit, timeUnit = _c === void 0 ? exports.UNITS.MS : _c, _d = _b.reject, reject = _d === void 0 ? false : _d;
    switch (timeUnit) {
        case exports.UNITS.D:
            time *= 24;
        case exports.UNITS.H:
            time *= 60;
        case exports.UNITS.M:
            time *= 60;
        case exports.UNITS.S:
            time *= 1000;
        default:
        // pass
    }
    return new Promise(function (res, rej) {
        setTimeout(reject ? rej : res, time);
    });
}
exports.wait = wait;
